### PR TITLE
adding option for data being the first promise parameter #112

### PR DESCRIPTION
**Pull request by abeadam**

### CODE REVIEW

**Comment by abeadam** [comment]
Happy to add tests for the case when selfParameter is true if the code looks good to a reviewer(s)

**Comment by guigs** [comment]
A similar problem happens with success callback. Should this PR address this case too?

**Comment by benkeen** [comment]
+1 on getting this sucker in. Seems like this PR should handle all scenarios where the libs passes in an extra param.

File: backbone.fetch-cache.js [hunk 1]
```diff
@@ -229,7 +231,7 @@
       // ...finish and return if we're not
       else {
         if (_.isFunction(opts.success)) { opts.success(self, attributes, opts); }
-        deferred.resolve(self);
+        deferred.resolve.apply(self, resolveValues);
```
Comment (dyurkavets) [inline backbone.fetch-cache.js]: `deferred.resolveWith`

**Pull request merged**